src/freva/utils.py

```diff
@@ -176,7 +182,9 @@ def _hist(self) -> Dict[str, Any]:
         try:
             hist = cast(
                 List[Dict[str, Any]],
-                freva.history(entry_ids=self._id, return_results=True),
+                freva.history(
+                    entry_ids=self._id, return_results=True, user_name=self._user_name
+                ),
```

Comment:
```suggestion
                freva.history(
                    entry_ids=self._id, return_results=True, user_name="*"
                ),
```
Remember you've build in the `user-name` feature and that users can query `all` users. I think we can make use of that feature here: The history ids are unique and since the user already knows the id of a run from another user there is not much need in telling the thing again what user was involved with this id. Look at this (I am logged on as me and use the xces module) :

```python
from functools import partial
freva.history=partial(freva.history, user_name="*") # Override the history command
status = freva.PluginStatus(933409)
status._hist["uid_id"]
'b380001'
``
So in think you simply have to set `user_name ` in the `freva.history` call to "*" or "all" and you are done. No need for another `user_name` parameter.

If you want you can introduce a property `user_name` or something that returns the  `uid_id` entry of the history.